other-tests/character-class.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 Mathias Bynens. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Mathias Bynens
+description: >
+  Unicode property escapes must be supported in character classes.
+esid: sec-static-semantics-unicodematchproperty-p
+features: [regexp-unicode-property-escapes]
+---*/
+
+/[\p{Hex}]/u;
+assert(
+  /[\p{Hex}-\uFFFF]/u.test('-'),
+  'property escape in character class should not be parsed as the start of a range'
+);
+assert.throws(SyntaxError, () => /[\p{}]/u);
```

Comment:
Why wouldn’t you be able to catch them?

```js
const isValid = (() => {
  try {
    Function('\\uGGGG'); // Note: you need to escape the source code here.
    return true;
  } catch (exception) {
    return false;
  }
})();

// `isValid` is `false`, because \uGGGG throws an early error
```

Could this technique be used in the tests? I’m fine with using negative tests, but in this case that would mean creating *lots* of different files. Some tests make more sense when grouped together, IMHO.